autoload/neomake/makers/ft/markdown.vim

```diff
@@ -1,33 +1,64 @@
 function! neomake#makers#ft#markdown#SupersetOf() abort
     return 'text'
 endfunction
+
 function! neomake#makers#ft#markdown#EnabledMakers() abort
     let makers = executable('mdl') ? ['mdl'] : ['markdownlint']
+    if executable('vale') | let makers += ['vale'] | endif
     return makers + ['writegood'] + neomake#makers#ft#text#EnabledMakers()
 endfunction
 
 function! neomake#makers#ft#markdown#mdl() abort
     return {
-        \ 'errorformat':
-        \   '%W%f:%l: MD%n %m,' .
+        \
+        \ 'errorformat': '%W%f:%l: MD%n %m,' .
         \   '%W%f:%l: %m',
         \ 'output_stream': 'stdout',
         \ }
 endfunction
 
 function! neomake#makers#ft#markdown#markdownlint() abort
     return {
-                \ 'errorformat':
-                \ '%f: %l: %m'
-                \ }
+        \ 'errorformat': '%f: %l: %m'
+        \ }
 endfunction
 
 function! neomake#makers#ft#markdown#alex() abort
     return {
-                \ 'errorformat':
-                \ '%P%f,' .
-                \ '%-Q,' .
-                \ '%*[ ]%l:%c-%*\d:%n%*[ ]%tarning%*[ ]%m,' .
-                \ '%-G%.%#'
-                \ }
+        \ 'errorformat':
+        \   '%P%f,'
+        \   .'%-Q,'
+        \   .'%*[ ]%l:%c-%*\d:%n%*[ ]%tarning%*[ ]%m,'
+        \   .'%-G%.%#'
+        \ }
+endfunction
+
+function! neomake#makers#ft#markdown#ProcessVale(context) abort
+    let bufnr = a:context['jobinfo'].bufnr
+    let filename = buffer_name(bufnr)
+    let items = get(a:context['json'], filename, [])
```

Comment:
I think you could just iterate of the filenames and use them as-is?
This way it would be ready to be used as a project maker, too - i.e. it would add entries for all buffers.
Filtering only on the current buffer would be more the responsibility of Neomake, not the maker.

Apart from that `buffer_name` is an obsolete name for `bufnr`.